packages/core/src/dialects/abstract/query-interface-typescript.ts

```diff
@@ -741,4 +745,82 @@ export class AbstractQueryInterfaceTypeScript<Dialect extends AbstractDialect =
 
     return this.sequelize.queryRaw(sql, { ...bulkDeleteOptions, raw: true, type: QueryTypes.DELETE });
   }
+
+  /**
+   * Update multiple records of a table
+   *
+   * @example
+   * queryInterface.bulkUpdate('roles', {
+   *     label: 'admin',
+   *   }, {
+   *     userType: 3,
+   *   },
+   * );
+   *
+   * @param tableName         Table name to update
+   * @param values            Values to be inserted, mapped to field name
+   * @param options           Various options, please see Model.bulkCreate options
+   * @param columnDefinitions Attributes on return objects if supported by SQL dialect
+   */
+  async bulkUpdate(
+    tableName: TableNameOrModel,
+    values: Record<string, unknown>,
+    options?: QiUpdateOptions,
+    columnDefinitions?: Record<string, AttributeOptions>,
+  ): Promise<number> {
+    if (options?.bind) {
+      assertNoReservedBind(options.bind);
+    }
+
+    const bulkUpdateOptions = { ...options };
+    const { bind, query } = this.queryGenerator.updateQuery(tableName, values, bulkUpdateOptions, columnDefinitions);
+
+    // unlike bind, replacements are handled by QueryGenerator, not QueryRaw
+    delete bulkUpdateOptions.replacements;
+    bulkUpdateOptions.bind = combineBinds(bulkUpdateOptions.bind ?? {}, bind ?? {});
+
+    return this.sequelize.queryRaw(query, { ...bulkUpdateOptions, type: QueryTypes.BULKUPDATE });
+  }
+
+  /**
+   * Updates a row
+   *
+   * @param tableName
+   * @param values
+   * @param options
+   * @param instanceOrColumnDefinitions
+   */
+  async update<M extends Model>(
```

Comment:
we should merge `update` and `bulkUpdate`:

The main difference between the two is:

- `update` uses `QueryTypes.update`, `bulkUpdate` uses `QueryTypes.bulkUpdate`. Here is their current behavior:
  - `QueryTypes.buldUpdate` only returns the update count
  - `QueryTypes.update` returns both the updated data and the update count.

  IMO this is a weird design and we should change them to do this:
  - `QueryTypes.bulkUpdate` should return `[updateCount: number, updatedRows: object[]]`
  - `QueryTypes.update` should return `[updateCount: number, updatedRow: object]` (+ it should log a warning in the console if `updateCount` > 1)

  The merged QueryInterface `update` & `bulkUpdate` methods should use `QueryTypes.bulkUpdate`. `QueryTypes.update` is meant for user-usage only.
- `update` produces a model instance, or can update a model instance. IMO this is a mistake. If someone needs to manipulate model instances, they should use model methods. The values returned by these two methods should be plain objects that contain the values from the database. The model method that call these two methods should then decide what to do with the data. `model.update` would merge it with its own, `Model.bulkUpdate` would do nothing with it.

With these changes, we could then drop `Qi.update` and only keep `Qi.bulkUpdate`

Reply:
See comment below